v03_pipeline/lib/reference_data/dataset_table_operations_test.py

```diff
@@ -15,10 +15,15 @@
     dbnsfp_mito_custom_select,
 )
 from v03_pipeline.lib.reference_data.dataset_table_operations import (
+    _ht_enums_match_config,
```

Comment:
This is kinda a me not loving underscore prefixes outside of classes thing, but what if we just moved these compare functions to a new module?  `dataset_table_operations` is getting big and these functions have a shared purpose.

Reply:
That's a good way to think about what should be a class. I was thinking of creating a `Validator` class that would have `joined_ht_globals` as a class attribute or something, but 
- it's not necessarily an improvement from the functional way I've written it now
- it doesn't follow the pattern you described of what actually should be an object
- it'd diverge from codebase standards
- it'd make moving my tests to a new file into a bigger refactor, which I actually don't want 
so I'll just move them and tweak the logic per your other comments

Reply:
Yep!  I think everything you've mentioned makes lots of sense.  I'd actually love for the dictionaries we have in the config (with the schema that's loosely defined in the big comment) and on the globals to be more defined (maybe something like a [pydantic](https://docs.pydantic.dev/latest/) model or a data class or something), which I think does pave the way for having a more class based approached.

Reply:
Yeah, with the current structure of `CONFIG`, rules like if `custom_import` exists then `source_path` should too aren't really enforced. I think pydantic is the right choice because it has nice syntax for writing separate validation functions with `@validator` decorator that enforce these rules.